bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/OpenWeatherMapWeatherAndForecastHandler.java

```diff
@@ -376,12 +385,12 @@ private void updateHourlyForecastChannel(ChannelUID channelUID, int count) {
                     state = getQuantityTypeState(forecastData.getClouds().getAll(), PERCENT);
                     break;
                 case CHANNEL_RAIN:
-                    state = getQuantityTypeState(
-                            forecastData.getRain() == null ? 0 : forecastData.getRain().getVolume(), MILLI(METRE));
+                    Rain rain = forecastData.getRain();
+                    state = getQuantityTypeState(rain == null ? 0 : rain.getVolume(), MILLI(METRE));
                     break;
                 case CHANNEL_SNOW:
-                    state = getQuantityTypeState(
-                            forecastData.getSnow() == null ? 0 : forecastData.getSnow().getVolume(), MILLI(METRE));
+                    Snow snow = forecastData.getSnow();
```

Comment:
See above.

Reply:
This is not doing anything, are you sure, I'm now passing 0 if its 0?